### PR TITLE
hash_value is not defined inside an #ifdef OM_HAS_HASH

### PR DESCRIPTION
**Pull request by afabri**
So we have to add it for OpenMesh version < 6.2  
@janetournois can you test this please.

### CODE REVIEW

File: STL_Extension/include/CGAL/hash_openmesh.h [hunk 1]
```diff
@@ -22,15 +22,20 @@
 
 #include <OpenMesh/Core/Mesh/Handles.hh>
 
-#ifndef OM_HAS_HASH
-
-#include <functional>
+#if OMVERSION < 0x60200
```
Comment (sloriot) [inline STL_Extension/include/CGAL/hash_openmesh.h]: On my OpenMesh installation, the variable is named `OM_VERSION`.

**Comment by janetournois** [comment]
it works for me, and fixes #1360

**Comment by lrineau** [comment]
It could help if the `examples/BGL_OpenMesh/CMakeLists.txt` displays the version of OpenMesh that is being used. I would like to know which OpenMesh versions are actually tested in our testsuite.

It seems the following test platforms have OpenMesh (at least in [CGAL-4.9-Ic-137]):
  - `jtournois_x64_Cygwin-Windows10_MSVC2015-Debug-64bits`
  - `jtournois_x64_Cygwin-Windows10_MSVC2015-Release-64bits`
  - `lrineau_Fedora-rawhide`
  - `lrineau_Fedora-strict-ansi`
  - `lrineau_x86-64_Linux-Fedora19_llvm-clang-with-g++-4.8`
  - `lrineau_x86-64_Linux-Fedora_IntelCompiler-17`
  - `lrineau_x86-64_Linux-Fedora_IntelCompiler-17-with-assertions`

[CGAL-4.9-Ic-137]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-137.shtml

I had a look and our current module `Installation/cmake/modules/FindOpenMesh.cmake` does not detect the OpenMesh version. It should read the file `${OPENMESH_INCLUDE_DIR}/OpenMesh/Core/System/config.h` and parse the line:
```C++
#define OM_VERSION 0x30200
```

**Comment by afabri** [comment]
Should it be in cmake, or should it be so that it can be written in the  table of the testsuite?

**Comment by lrineau** [comment]
Andreas in https://github.com/CGAL/cgal/pull/1377#issuecomment-243709612 said:
> Should it be in cmake, or should it be so that it can be written in the table of the testsuite?

One or the other, but I think the CMake way is simpler: modify `FindOpenMesh.cmake` to parse the `OM_VERSION` from `<OpenMesh/Core/System/config.h>`, then modify `examples/BGL_OpenMesh/CMakeLists.txt` so that the version string is displayed.

For the other way, first OpenMesh should be searched in the configuration of CGAL libraries, even if that is not at all a dependency of CGAL libraries, and then we would have to create a .cpp file to display the version string, and modify also two or three scripts from the testsuite infrastructure, so that in the end the version string is displayed in the table of the testsuite results.

**Pull request merged**